dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/data/upload/v2/DataUploadRunnable.kt

```diff
@@ -54,28 +55,40 @@ internal class DataUploadRunnable(
             val context = contextProvider.context
             // TODO RUMM-0000 it should be already on the worker thread and if readNextBatch is async,
             //  we should wait until it completes before scheduling further
-            val lock = CountDownLatch(1)
-            storage.readNextBatch(
-                noBatchCallback = {
-                    increaseInterval()
-                    lock.countDown()
+            var batchesSent = 0
+            var lastUploadWasSuccess = false
+            do  {
+                val lock = CountDownLatch(1)
+                storage.readNextBatch(
+                    noBatchCallback = {
+                        increaseInterval()
+                        lock.countDown()
+                        batchesSent = maxBatchesPersUpload
```

Comment:
this line is misleading and not actually true, the line with `lastUploadWasSuccess` below should be enough. Btw, is it possible to just use `break` here?

Reply:
yes a break would have been nice, unfortunately the compiler complains about using a break outside of the function/class :(. I wonder if there is any workaround for this in Kotlin, I could not find any except using a flag.